crates/binstalk-downloader/src/remote/resolver.rs

```diff
@@ -41,7 +46,19 @@ fn get_configs() -> Result<(ResolverConfig, ResolverOpts), BoxError> {
     let mut config = ResolverConfig::new();
     let opts = ResolverOpts::default();
 
-    get_adapter()?.dns_servers().iter().for_each(|addr| {
+    let interface = get_default_interface()?;
+
+    if interface.dns_servers.is_empty() {
+        warn!("No DNS servers found on default interface; falling back to system DNS config");
+
+        if let Ok((sys_config, sys_opts)) = system_conf::read_system_conf() {
+            return Ok((sys_config, sys_opts));
+        }
+        
+        return Err("No DNS servers found and failed to read system DNS config".into());
```

Comment:
It'd be great to have the error 
`system_conf::read_system_confg` included in the error message, or maybe just propagate the error?